app/src/swap.c

```diff
@@ -0,0 +1,290 @@
+/*******************************************************************************
+*   (c) 2016 Ledger
+*   (c) 2018 - 2023 Zondax AG
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+********************************************************************************/
+#include "swap.h"
+#include "crypto.h"
+#include "bignum.h"
+#include "zxformat.h"
+#include "substrate_dispatch.h"
+
+static zxerr_t extractHDPath(uint8_t* params, uint8_t paramsSize);
+static zxerr_t readU32BE(uint8_t* input, uint32_t *output);
+static zxerr_t readU64BE(uint8_t* input, uint64_t *output);
+static zxerr_t bytesAmountToStringBalance(uint8_t *amount, uint8_t amount_len, char *out, uint8_t out_len);
+
+static void handle_check_address(check_address_parameters_t* params);
+static void handle_get_printable_amount(get_printable_amount_parameters_t* params);
+
+static bool copy_transaction_parameters(create_transaction_parameters_t* sign_transaction_params);
+
+swap_globals_t G_swap_state;
+
+void swap_handle_command(libargs_s *args) {
+    switch (args->command) {
+        case check_address:
+            handle_check_address(args->check_address);
+            G_swap_state.called_from_swap = 0;
```

Comment:
we can set this line before entering the switch statement

Reply:
@ftheirs we need to set the global variable after calling the command handling functions to avoid potential bss section memory overlap issues with the arguments passed by app-exchange during the sideloading.

If you set the variable before, you might wipe out the arguments inadvertently (and it happened in some tests)